### ISSUE TITLE
test_loader.FileTest.test_filenameNotPython should create its test data

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6430]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6430|
|Type|task|
|Created|2013-04-12 16:45:10Z|
|Branch|https://github.com/twisted/twisted/tree/test_filenameNotPython-testdata-6430|

`trial.test.test_loader.FileTest.test_filenameNotPython` uses a static text file ([twisted/trial/test/notpython](twisted/trial/test/notpython)) but it might as well be a file thats being created, using `FilePath`, during the test run.

Attachments:

* [6430.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/dbe/dbe98bfe8cad58e02d9ea22fc98f446240edc909/7c64ede8052332381c2faf39c962534573886741.patch) (707 bytes) - added by introom on 2013-04-15 16:13:24Z - 
* [6430_2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/dbe/dbe98bfe8cad58e02d9ea22fc98f446240edc909/5c1832a220eaf4d7e11bf8b9b228bb9d7b613af1.patch) (1101 bytes) - added by introom on 2013-04-17 11:57:39Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6430 6430
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_test_filenameNotPython_testdata_6430 branches/test_filenameNotPython-testdata-6430
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__easy easy
time__1365785110000000 1365785110000000
changetime__1366230665000000 1366230665000000
version__None None
owner__tom_prince tom.prince
cc__thijs cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|introom commented|
|-|-|

created a file named "notpython2" on the fly.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince set owner to introom|
|-|-|

Thanks for your contribution.

 1. You add code that tests against a created file, but leave the code that tests against the file in the source tree. Here are some comments about why we only want the former:
```
<exarkun> keeping data in the source tree should be avoided where possible
<exarkun> I don't think anyone knows how to avoid it in _all_ cases, but where it's obvious how to do so it seems worthwhile.
<exarkun> And dumping data into a temporary path during the test run opens up more testing opportunities
<exarkun> If the data is part of the shipped source, then it's hard to rely on some of the less obvious properties (permissions, for example)
<exarkun> There's a handful of tests that fail when run against an installed copy of Twisted.  I dunno if this test is one of them, but it suffers from the same problem as the failing tests do, even if it doesn't have the same symptom
```
 2. Also, the test data in the source tree that will no longer be used should be removed (after checking that nothing else is using it.
 3. since the file is created in a temporary directory, the file propbably doesn't need to be removed. But if you do remove it, you should ensure that it is always removed.
 4. This change needs a [newsfile](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles) (probably `.misc` in this case).

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|introom commented|
|-|-|

Replying to [tom.prince](#note_3):
> Thanks for your contribution.
> 
>  1. You add code that tests against a created file, but leave the code that tests against the file in the source tree. Here are some comments about why we only want the former:

I don't quite understand what "leave the code that tests against the file in the source tree" means.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince commented|
|-|-|

That should've been "left" probably.

This ticket is about creating the data in the test *instead of* using the test data in the tree. Your patch creates the data *in addition to* using the test data in the tree

**Comment by twisted-trac** [comment]
|<img alt="introom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|introom removed owner|
|-|-|

Replying to [tom.prince](#note_5):
> That should've been "left" probably.
> 
> This ticket is about creating the data in the test *instead of* using the test data in the tree. Your patch creates the data *in addition to* using the test data in the tree

Thanks for your comment.
I removed the original assertion against the notpython file and replaced it with a generated file.
Also, I did a "ack notpython" in the trunk, and found only this test uses the "notpython" file. So it's safe to delete the "notpython" file.

I also added an empty .misc topfile. (I guess because it's empty so it's not shown up in this trac system)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince commented|
|-|-|

(In [38119]) Branching to test_filenameNotPython-testdata-6430.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince set owner to @tomprince|
|-|-|

This looks good. Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_9"></a>|@tomprince set status to `closed`|
|-|-|

(In [38120]) Merge test_filenameNotPython-testdata-6430: Make test_loader.FileTest.test_filenameNotPython create its test data.

Author: introom
Reviewers: tom.prince
Fixes: [#6430](https://github.com/twisted/twisted/issues/6430)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

The file is still [twisted/trial/test/notpython there](twisted/trial/test/notpython there), only it's contents was erased.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince set status to `reopened`|
|-|-|

(In [38127]) Revert r38120: File not actually deleted.

Reopens: [#6430](https://github.com/twisted/twisted/issues/6430)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_12"></a>|@tomprince commented|
|-|-|

(In [38128]) Apply 6430_2.patch from introom.

Refs: [#6430](https://github.com/twisted/twisted/issues/6430)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_13"></a>|@tomprince set owner to @thijstriemstra<br>@tomprince set status to `new`|
|-|-|

Thanks for catching that. It should be fixed now.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set owner to @tomprince|
|-|-|

Thanks, looks good.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_15"></a>|@tomprince set status to `closed`|
|-|-|

(In [38130]) Merge test_filenameNotPython-testdata-6430: Make test_loader.FileTest.test_filenameNotPython create its test data.

Author: introom
Reviewers: tom.prince
Fixes: [#6430](https://github.com/twisted/twisted/issues/6430)

Now with the file actually deleted.
